packages/background/src/infrastructure/stores/migrator/migrations/migration-52.ts

```diff
@@ -0,0 +1,98 @@
+import { BlankAppState } from '@block-wallet/background/utils/constants/initialState';
+import { IMigration } from '../IMigration';
+import {
+    SLOW_TESTNET_TIME_INTERVALS_DEFAULT_VALUES,
+    ACTIONS_TIME_INTERVALS_DEFAULT_VALUES,
+} from '../../../../utils/constants/networks';
+import { FEATURES } from '../../../../utils/constants/features';
+import {
+    DEFAULT_TORNADO_CONFIRMATION,
+    DERIVATIONS_FORWARD,
+} from '../../../../controllers/blank-deposit/types';
+import { normalizeNetworksOrder } from '../../../../utils/networks';
+
+/**
+ * This migration adds RSK Mainnet and RSK Testnet
+ */
+export default {
+    migrate: async (persistedState: BlankAppState) => {
+        const { availableNetworks } = persistedState.NetworkController;
+        const updatedNetworks = { ...availableNetworks };
```

Comment:
We have to take into account that the user may already have added this network to his wallet. So, we should first look for the RSK chain ID in the user's network list and if it exists:
- Delete the user's RSK network record, indexed as `CHAIN-30`
- Update some of our parameters with the user's configured network (like the RPC url, name, order, so on...)

Reply:
You can use the `getNonNativeNetworkKey` helper to get the Key